src/main/java/com/fpl/smdc/core/notification/config/AWSAppConfig.java

```diff
@@ -0,0 +1,144 @@
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.AmazonSNSClientBuilder;
+
+import io.github.gilbertojrequena.bonsai_sns.server.BonsaiSnsEnvironment;
+import io.github.gilbertojrequena.bonsai_sns.server.BonsaiSnsServer;
+import io.github.gilbertojrequena.bonsai_sns.server.Subscription;
+import io.github.gilbertojrequena.bonsai_sns.server.Topic;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.aws.messaging.core.NotificationMessagingTemplate;
+import org.springframework.cloud.aws.messaging.endpoint.NotificationMessageHandlerMethodArgumentResolver;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+
+@Configuration
+@PropertySource("classpath:application.properties")
+//@EnableWebMvc
+public class AWSAppConfig extends WebMvcConfigurerAdapter{
+
+    private static final Integer TEMPORARY_CREDENTIALS_DURATION_DEFAULT = 7200;
+
+    @Value("${aws.temporary.credentials.validity.duration}")
+    String credentialsValidityDuration;
+
+    @Value("${aws.sns.topic.demo.ARN}")
+    String snsTopicDemoARN;
+
+    @Bean(name = "snsTopicDemoARN")
+    public String snsTopcARN() {
+        return this.snsTopicDemoARN;
+    }
+
+    @Value("${sns.topic.arn}")
+    private String snsTopicARN;
+
+    @Value("${aws.accessKey}")
+    private String awsAccessKey;
+
+    @Value("${aws.secretKey}")
+    private String awsSecretKey;
+
+    @Value("${aws.region}")
+    private String awsRegion;
+
+    @SuppressWarnings("deprecation")
+	@Bean(name = "sessionCredentials")
+    public BasicAWSCredentials sessionCredentials() {
+
+        AWSSecurityTokenServiceClient sts_client = (AWSSecurityTokenServiceClient) AWSSecurityTokenServiceClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
+       // sts_client.withRegion(Regions.US_EAST_1);
+        GetSessionTokenRequest session_token_request = new GetSessionTokenRequest();
+      
+        if (this.credentialsValidityDuration == null || this.credentialsValidityDuration.trim().equals("")) {
+            session_token_request.setDurationSeconds(TEMPORARY_CREDENTIALS_DURATION_DEFAULT);
+        } else {
+            session_token_request.setDurationSeconds(Integer.parseInt(this.credentialsValidityDuration));
+        }
+
+        GetSessionTokenResult session_token_result = sts_client.getSessionToken(session_token_request);
+
+        Credentials session_creds = session_token_result.getCredentials();
+
+        BasicAWSCredentials  sessionCredentials = new BasicAWSCredentials (
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).